src/libraries/Common/src/System/Security/Cryptography/RsaPaddingProcessor.cs

```diff
@@ -10,62 +10,62 @@ namespace System.Security.Cryptography
-        private static ReadOnlySpan<byte> DigestInfoMD5 => new byte[]
-            {
+        private static ReadOnlySpan<byte> DigestInfoMD5 =>
+            [
                 0x30, 0x20, 0x30, 0x0C, 0x06, 0x08, 0x2A, 0x86,
                 0x48, 0x86, 0xF7, 0x0D, 0x02, 0x05, 0x05, 0x00,
                 0x04, 0x10,
-            };
+            ];
 
-        private static ReadOnlySpan<byte> DigestInfoSha1 => new byte[]
-            {
+        private static ReadOnlySpan<byte> DigestInfoSha1 =>
+            [
                 0x30, 0x21, 0x30, 0x09, 0x06, 0x05, 0x2B, 0x0E, 0x03,
                 0x02, 0x1A, 0x05, 0x00, 0x04, 0x14,
-            };
+            ];
 
-        private static ReadOnlySpan<byte> DigestInfoSha256 => new byte[]
-            {
+        private static ReadOnlySpan<byte> DigestInfoSha256 =>
+            [
                 0x30, 0x31, 0x30, 0x0D, 0x06, 0x09, 0x60, 0x86, 0x48,
                 0x01, 0x65, 0x03, 0x04, 0x02, 0x01, 0x05, 0x00, 0x04,
                 0x20,
-            };
+            ];
 
-        private static ReadOnlySpan<byte> DigestInfoSha384 => new byte[]
-            {
+        private static ReadOnlySpan<byte> DigestInfoSha384 =>
+            [
                 0x30, 0x41, 0x30, 0x0D, 0x06, 0x09, 0x60, 0x86, 0x48,
                 0x01, 0x65, 0x03, 0x04, 0x02, 0x02, 0x05, 0x00, 0x04,
                 0x30,
-            };
+            ];
 
-        private static ReadOnlySpan<byte> DigestInfoSha512 => new byte[]
-            {
+        private static ReadOnlySpan<byte> DigestInfoSha512 =>
+            [
                 0x30, 0x51, 0x30, 0x0D, 0x06, 0x09, 0x60, 0x86, 0x48,
                 0x01, 0x65, 0x03, 0x04, 0x02, 0x03, 0x05, 0x00, 0x04,
                 0x40,
-            };
+            ];
 
-        private static ReadOnlySpan<byte> DigestInfoSha3_256 => new byte[]
-            {
+        private static ReadOnlySpan<byte> DigestInfoSha3_256 =>
+            [
                 0x30, 0x31, 0x30, 0x0D, 0x06, 0x09, 0x60, 0x86, 0x48,
                 0x01, 0x65, 0x03, 0x04, 0x02, 0x08, 0x05, 0x00, 0x04,
                 0x20,
-            };
+            ];
 
-        private static ReadOnlySpan<byte> DigestInfoSha3_384 => new byte[]
-            {
+        private static ReadOnlySpan<byte> DigestInfoSha3_384 =>
+            [
                 0x30, 0x41, 0x30, 0x0D, 0x06, 0x09, 0x60, 0x86, 0x48,
                 0x01, 0x65, 0x03, 0x04, 0x02, 0x09, 0x05, 0x00, 0x04,
                 0x30,
-            };
+            ];
 
-        private static ReadOnlySpan<byte> DigestInfoSha3_512 => new byte[]
-            {
+        private static ReadOnlySpan<byte> DigestInfoSha3_512 =>
+            [
                 0x30, 0x51, 0x30, 0x0D, 0x06, 0x09, 0x60, 0x86, 0x48,
                 0x01, 0x65, 0x03, 0x04, 0x02, 0x0A, 0x05, 0x00, 0x04,
                 0x40,
-            };
+            ];
 
-        private static ReadOnlySpan<byte> EightZeros => new byte[8];
```

Comment:
Note that this one was actually allocating on each call :(

Reply:
Is that necessarily bad? I know you and others have pushed back on caching memory that would never leave Gen0 (I don't know this code OC)

Reply:
> Is that necessarily bad? Often we avoid caching memory that would never leave Gen0 (I don't know this code OC)

If this had instead been:
```C#
private static ReadOnlySpan<byte> EightZeros => new byte[8] { 0, 0, 0, 0, 0, 0, 0, 0 };
```
there wouldn't have been any allocation, not even on first use: it would have simply been eight zero'd bytes in the assembly data, and the span would have pointed directly to that memory.
[SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPgYhgVwDaosVBTABAEygGdMcBYAKAAEAmARkqoGZcq6A2Vm3AYVwG9KuYbgAOAJygA3CABc8bTgCVsEfAHkYqAJ4BlURBgAeYNvkA+XAEFcAXksxsAd1yn5AbQAcAXQDcQkQlpOQUOXBU1TR19QxMzbEsAITsHZ1d4r28BXAAGABpcgvzCkuKy3ABffwoKoA)

And, yes, a property that looks like it should be non-allocating but that allocates on every access is bad :)

Reply:
Oh, wow, TIL.

Reply:
We should just fix that in Roslyn. I don't see a good reason for these two to exhibit such a codegen difference. (Though moving forward we'll use that syntax much less and so hopefully won't trip over it again.)
cc: @jcouv